### ISSUE TITLE
`amend` panics if the branch hasn't been pushed upstream

### ISSUE THREAD

**Issue by abatilo** [ejoffe/spr#115]
```
git error: There is no tracking information for the current branch.
Please specify which branch you want to rebase against.
See git-rebase(1) for details.

    git rebase <branch>

If you wish to set tracking information for this branch you can do so with:

    git branch --set-upstream-to=origin/<branch> abatilo/testing-spr

panic: exit status 1

goroutine 1 [running]:
github.com/ejoffe/spr/spr.check(...)
        /Users/eitan/Code/spr/spr/spr.go:426
github.com/ejoffe/spr/spr.(*stackediff).mustgit(...)
        /Users/eitan/Code/spr/spr/spr.go:421
github.com/ejoffe/spr/spr.(*stackediff).AmendCommit(0xc000139f28, 0x5efa78, 0xc00001c110)
        /Users/eitan/Code/spr/spr/spr.go:73 +0x6b0
main.main()
        /Users/eitan/Code/spr/cmd/amend/main.go:57 +0x32e
```

When I tried to amend a commit, it panicked when the branch I created was not yet pushed to the remote.

**Comment by ejoffe** [comment]
@abatilo  - thanks for bringing up all the interesting cases.
what version are you running?

**Comment by abatilo** [comment]
No problem! Thanks for the project!

I installed via downloading the v0.7.2 release on the GitHub releases page.

I may try to tackle some of these. One of the reasons why I wanted to try this project is because I like writing Go so a Go project means there's an opportunity for me to contribute!

**Comment by ejoffe** [comment]
Great!
For this particular issue, I think there is a recent fix: 
`0a5b154 fix git amend to rebase locally instead of origin/master`
It's in version 0.7.12, you can try to reproduce and see if it happens. 
The fix makes git amend - rebase locally instead of on top of the remote branch, which is the correct behavior, and I think should alleviate the issue you had. 
I'll comment in the other issues, to maintain context.

**Comment by hasit** [comment]
Hello, I still see this issue on v0.7.15

```
$ git-spr version
0.7.15 : 2021-11-19T07:28:25Z : 3439ad32

$ git amend
 2 : 20fce708 : some commit message
 1 : 1522d5da : commit message
Commit to amend [1-2]: 2
git error: There is no tracking information for the current branch.
Please specify which branch you want to rebase against.
See git-rebase(1) for details.

    git rebase '<branch>'

If you wish to set tracking information for this branch you can do so with:

    git branch --set-upstream-to=<remote>/<branch> some-branch

panic: exit status 1

goroutine 1 [running]:
github.com/ejoffe/spr/spr.check(...)
	/Users/runner/work/spr/spr/spr/spr.go:444
github.com/ejoffe/spr/spr.(*stackediff).mustgit(...)
	/Users/runner/work/spr/spr/spr/spr.go:439
github.com/ejoffe/spr/spr.(*stackediff).AmendCommit(0xc0000c3f10, {0xc00000e3f0, 0x12})
	/Users/runner/work/spr/spr/spr/spr.go:76 +0x525
main.main()
	/Users/runner/work/spr/spr/cmd/amend/main.go:57 +0x32f
```

**Comment by yogurtearl** [comment]
Seeing similar on 0.8.3

**Comment by ejoffe** [comment]
Fixed in v0.8.4
